FLOE-586: Migrate FLOE Project to 11ty static site generator (GSOC 2020)

sachin10101998 opened the pull request:
Migrated Floe Project source code to 11ty as part of the GSOC 2020 project.

Fixes #170 .
Fixes #182.

sachin10101998 commented:
@jobara @colinbdclark @jhung @BlueSlug @greatislander I migrated floe to 11ty. I have updated README.md with steps to run it locally for testing. Initially, I added the Github actions workflow which ran as expected but I removed it because of differences between branch names of floe project and my fork. I'd be updating the CI/CD after this gets merged. Please review.

jobara commented:
@jhung lets do a review of the open PRs agains the floeproject.org repo. We should sort those out before merging this.

jobara commented on Gruntfile.js:
```diff
@@ -0,0 +1,136 @@
+    grunt.initConfig({
+        // Project package file destination.
+        pkg: grunt.file.readJSON("package.json"),
+        eslint: {
+            all: ["**/*.js"]
+        },
+        jsonlint: {
+            all: ["package.json", ".eslintrc.json"]
+        },
```
Please replace these with [gpii-grunt-lint-all](https://www.npmjs.com/package/gpii-grunt-lint-all).

sachin10101998 replied:
Fixed

jobara commented on package.json:
```diff
@@ -0,0 +1,55 @@
+  "version": "1.0.0",
+  "description": "FLOE Project Website",
+  "main": "index.js",
+  "scripts": {
+    "build": "eleventy",
+    "eleventy": "eleventy --serve",
+    "eleventyport": "eleventy --serve --port=9778"
+  },
```
Should probably add a script for linting. Should add a [`prepare`](https://docs.npmjs.com/misc/scripts) for loading the front end dependencies. That will run on a local `npm install`.

sachin10101998 replied:
Fixed

jobara commented on package.json:
```diff
@@ -0,0 +1,55 @@
+  "contributors": [
+    "Sachin Chopra <schopra@mt.iitr.ac.in> (www.github.com/sachin10101998)"
+  ],
+  "license": "ISC",
```
The license should be [Creative Commons BY 4.0](https://creativecommons.org/licenses/by/4.0/). Here's an example of a [license file](https://github.com/fluid-project/.github/blob/master/LICENSE.md
) that should also be included in the repository.

jobara commented on package.json:
```diff
@@ -0,0 +1,55 @@
+    "Education"
+  ],
+  "author": "OCAD University <idrc@ocadu.ca>",
+  "contributors": [
+    "Sachin Chopra <schopra@mt.iitr.ac.in> (www.github.com/sachin10101998)"
+  ],
```
You should remove the contributors section. Mostly it's just hard to maintain. We've started using a centralized AUTHORS.md file and copyright. See an example from Infusion:

- https://github.com/fluid-project/infusion/blob/master/AUTHORS.md
- https://github.com/fluid-project/infusion/blob/master/Gruntfile.js

sachin10101998 replied:
Fixed

jobara commented on package.json:
```diff
@@ -0,0 +1,55 @@
+    "url": "https://github.com/fluid-project/floeproject.org/issues"
+  },
+  "homepage": "https://github.com/fluid-project/floeproject.org#readme",
+  "dependencies": {
+    "@11ty/eleventy": "^0.11.0",
+    "@11ty/eleventy-plugin-rss": "^1.0.7",
+    "@11ty/eleventy-plugin-syntaxhighlight": "^3.0.1",
+    "@tbranyen/jsdom": "^13.0.0",
+    "concurrently": "^5.2.0",
+    "grunt": "^1.1.0",
+    "grunt-contrib-clean": "^2.0.0",
+    "grunt-contrib-copy": "^1.0.0",
+    "html-minifier": "^4.0.0",
+    "image-size": "^0.8.3",
+    "infusion": "^3.0.0-dev.20200326T173810Z.24ddb2718",
+    "sanitize.css": "^11.0.1",
+    "slugify": "^1.4.0"
+  },
```
Please review if these are actually dev dependencies. That is are they only needed for testing, building and etc. Even for things that are included in the final output, but are copied over, like Infusion, could probably be moved to a dev dependency.

sachin10101998 replied:
Fixed

jobara commented on src/404.html:
```diff
@@ -0,0 +1,8 @@
+permalink: 404.html
+---
+<h1>{{ title }}</h1>
+<p>The page you were looking for does not exist yet. Go back to <a href="{{ '/' | url }}">home page?</a></p>
```
@jhung do you have suggestions for the 404 page both in terms of title and content. Do we want something that looks interesting or just something basic?

sachin10101998 replied:
@BlueSlug could you too weigh in on this?

BlueSlug replied:
I'd wait to see what @jhung says, though I would suggest something along the lines of: "The page you're looking for does not exist. Here are some helpful starting points: ...." (list the home page, News, etc.)

You could also try a friendlier message like "We can't seem to find the content you're looking for", or a cheeky one like "This isn't the content you're looking for". I think aiming for clarity is going to be the best bet, regardless, and anything is better than what's there now, which is a default Nginx error page.

jhung replied:
Something basic is fine. We'll probably end up revisiting this with a redesign anyway.

I would change the title to "Page not found" instead of "Oops! Not Found".

sachin10101998 replied:
Updated.

jobara commented on src/_data/global.js:
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    random() {
+      const segment = () => {
+        return (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
+      };
+      return `${segment()}-${segment()}-${segment()}`;
+    },
+    now: Date.now()
+  };
```
What are these used for?

sachin10101998 replied:
Removed. Not being currently used function.

jobara commented on src/_data/site.json:
```diff
@@ -0,0 +1,9 @@
+  "name": "FLOE Project",
+  "shortDesc": "Flexible Learning for Open Education",
+  "url": "https://www.floeproject.org",
+  "authorEmail": "schopra@mt.iitr.ac.in",
+  "authorHandle": "@sachin10101998",
+  "authorName": "Sachin Chopra",
```
Probably better to use the org info here.

sachin10101998 replied:
Fixed

jobara commented on src/_includes/layouts/default.html:
```diff
@@ -0,0 +1,55 @@
+        {% include partials/foundationflexgrid.html %}
+        
+
+        <!--[if lte IE 8]>
+        <script type="text/javascript">document.createElement("header")</script>
+        <script type="text/javascript">document.createElement("nav")</script>
+        <script type="text/javascript">document.createElement("footer")</script>
+        <script type="text/javascript">document.createElement("section")</script>
+        <script type="text/javascript">document.createElement("aside")</script>
+        <![endif]-->
```
I think we can remove all these IE8 fallbacks now.

sachin10101998 replied:
Fixed

colinbdclark commented on package.json:
```diff
@@ -0,0 +1,49 @@
+    "Education"
+  ],
+  "author": "OCAD University <idrc@ocadu.ca>",
+  "license": "CREATIVE COMMONS ATTRIBUTION 4.0 INTERNATIONAL LICENSE",
```
The SPDX license string for this license is "CC-BY-4.0".

sachin10101998 replied:
Fixed

sachin10101998 commented on src/_data/site.json:
```diff
@@ -0,0 +1,9 @@
+    "authorName": "OCAD University",
+    "authorWebsite": "https://www.ocadu.ca/",
+    "faviconPath": "{{ '/' | url }}assets/images/favicon.ico",
+    "title": "FLOE Project - Flecible Learning for open educaation"
```
Fixed

sachin10101998 commented on src/index.html:
```diff
@@ -0,0 +1,207 @@
+title: Paving the way toward inclusive Open Education Resources | floe
+---
+
+<div id="content" class="floe-content floe-story">
```
Sure.

sachin10101998 commented:
@jhung @BlueSlug  all requested changes have been made. Can we merge this now?

jhung commented on src/posts/index.md:
```diff
@@ -0,0 +1,23 @@
+    <div class="floe-news-archive">
+        <ul id="reverseneed">
+            {%- for post in collections.post -%}
+              <li id="reverseneed"><a href="{{ '/' | url }}{{ post.data.permalink }}"><p> {{ post.data.title }}</p></a> 
```
If using keyboard Tab key to navigate the post index, focus moves from the bottom of the page toward the top. This is a problem caused by the CSS transform rotation but the DOM order remaining the same. Ideally the DOM order should be consistent with Tab focus order so that people using only keyboards or switches (or other forms of navigation) don't get confused by focus moving backward. 

To fix this, I can see 2 ways:
1. When creating the Post collection, ensure the contents are sorted in reverse date order, or
2. In the loop traverse the collection in reverse so that the last collection item is first in the `ul` list.

Option 1 is the preferred method as it keeps the logic for data handling in the controller, leaving the template to just render the contents. Option 2 is a bit of a hack and should only be used if Option 1 isn't possible for some reason.

Side comment: `id` values should be unique in a document. The way `id` is being used here should really be a `class` instead. But if we use Option 1 or 2 above, we can get rid of these `id`s and remove the rotation CSS as it won't be needed anymore.

sachin10101998 replied:
fixed

jhung commented on src/posts/2019-10-23-SJRK-F2F.md:
```diff
@@ -0,0 +1,30 @@
+                    with youth at the margins, enabling them to not only tell their stories but
+                    also to capture and present them in ways they want to.
+                </p>
+                <p>
+                    <img src="images/SJRK_Drawing.png" alt="Collaborative surprise drawings by the SJRK team." /><br/>
+
+                    At the end of the day, the project team created a collaborative surprise drawing which represents
+                    the creative collaborations that happened throughout the project.
+                </p>
```
There's a strange bug here where part of the text content is appearing in a `<pre><code>` block.

![image](https://user-images.githubusercontent.com/564762/87324467-1f0e7580-c4fe-11ea-99f0-b7eab1906f9c.png)

sachin10101998 replied:
Fixed

jhung commented:
@sachin10101998 - does this properly build a Docker container?
@gtirloni, are there any specific requirements we need to do this?

gtirloni commented:
> @gtirloni, are there any specific requirements we need to do this?

@sachin10101998 please take a look at the following files in this PR https://github.com/fluid-project/fluidproject.org/pull/56/commits

* Dockerfile
* .dockerignore
* README (instructions to build Docker)
* .github/workflows

jhung commented on src/_includes/layouts/post.html:
```diff
@@ -0,0 +1,31 @@
+        </div>
+        <div class="floe">
+            {% include partials/header.html %}
+            {{content}}
```
The following should be added to the template so that it does not have to be duplicated for every news article.

```
<article class="floe-content floe-news-item">
   <h2> News </h2>
       <h3>{{ post.data.title }} </h3>
                <time class="floe-date" datetime="2012-03-06">{{ post.data.date }}</time>
{{content}}
</article>
```

sachin10101998 replied:
Fixed

jhung commented on src/posts/2012-03-06-webinar-outside-in.md:
```diff
@@ -0,0 +1,20 @@
+tags: post
+permalink: posts/2012-03-06-webinar-outside-in.html
+---
+<article class="floe-content floe-news-item">
+                <h2> News </h2>
+                <h3>Webinar: Outside In - Achieving Our Goals by Designing for Diversity</h3>
+                <time class="floe-date" datetime="2012-03-06">March 06, 2012</time>
```
Each article can have lines 8 to 11 refactored and put into the `post.html` template

sachin10101998 replied:
Done

sachin10101998 commented on README.md:
```diff
@@ -1,26 +1,30 @@
-What is the Floe Project?
-=========================
+# Fluid Project Website
```
fixed

sachin10101998 commented on src/posts/YYYY-MM-DD-newsarticletemplate.md:
```diff
@@ -0,0 +1,8 @@
+---
+layout: layouts/post
+title: Title of News Article
+date: '2000-01-01'
+tags: post
+permalink: posts/YYYY-MM-DD-newsArticleName.html
+---
+//Content Data
```
I have edited that file to provide template for the current news posts.

jhung commented:
@sachin10101998 There are conflicts now since Master changed on Wednesday (many of the news items were changed in Master and will need to be deleted again from this PR).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/fluid-project/floeproject.org?pullRequest=163) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/fluid-project/floeproject.org?pullRequest=163) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: sachin10101998<br/>:x: sanchita141011<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/fluid-project/floeproject.org?pullRequest=163) it.</sub>

jhung commented:
@sachin10101998 

Some more feedback:
1. Can the `posts` directory be renamed to `news` so the old URL structure still works (otherwise old links to news will receive a 404 message).
2. The news posts markdown `.md` files should use as much markdown as possible. Most HTML markup should be removed (only custom HTML not handled by markdown should remain). 
3. This pull request still has conflicts and can't be merged until that is addressed.

Thanks!

sachin10101998 commented:
@jhung  I have re configured the permalinks and solved the merge conflicts. Will update markdown files after discussion with you and @BlueSlug tomorrow.

jhung commented:
@sachin10101998  - The news URLs is still not what is needed.

For example, the URL structure as seen on the current production site is:
`https://floeproject.org/news/2018-08-30-Cities-Co-Design.html`

The URL structure in this pull request for news items should be the same. For example, `https://localhost:9778/news/2018-08-30-Cities-Co-Design.html` should go to the same news post.

The news files should be in a directory called `news` not `posts`.

Also please manually test the site before pushing commits to catch any errors.

jhung commented on src/_data/site.json:
```diff
@@ -0,0 +1,9 @@
+    "name": "FLOE Project",
+    "shortDesc": "Flexible Learning for Open Education",
+    "url": "https://www.floeproject.org",
+    "authorName": "OCAD University",
+    "authorWebsite": "https://www.ocadu.ca/",
```
Author Name should be `Inclusive Design Research Centre, OCAD University`
Author website should be `https://idrc.ocadu.ca`

sachin10101998 replied:
fixed

jhung commented on src/posts/2012-02-28-updates-to-ildh.md:
```diff
@@ -0,0 +1,24 @@
+layout: layouts/post
+title: Updates to the Inclusive Learning Design Handbook
+date: '2012-02-28'
+tags: post
```
What are tags and how are they being used in this instance of 11ty? Is the `post` tag being used for a function?

sachin10101998 replied:
They are used to make a collection which is then being used to render all the news articles on the news index page.The code which is using the tags is : 
    {%- for post in collections.post reversed -%}
    <li><a href="{{ '/' | url }}{{ post.data.permalink }}"><p> {{ post.data.title }}</p></a>
    <time class="floe-date" datetime="{{ post.data.date | w3DateFilter }}">{{ post.data.date | dateFilter }}</time>
    </li>
    {%- endfor -%}

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+      `---`
+      `Content Data`
+
+If you still have any qeries/doubts regarding the project, feel free to join our IRC Channel #fluid-work on Freenode.
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -1,18 +1,39 @@
+
+# To deploy to a personal webserver
 
+<<<<<<< HEAD
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -1,18 +1,39 @@
+<<<<<<< HEAD
+1. Run: `npm run build`
+2. Copy the contents of `dist` directory to your server.
+=======
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -32,12 +53,13 @@ If you make changes to the HTML code, repeat the steps to build the image and st
 
 News Item Template
 ==================
+>>>>>>> master
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+for the full diversity of learners and content producers.
+
+
+The present version of FLOE Project uses 11ty.
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+* `.gitignore` - List of folders not to be uploaded to Github. contains node_modules and dist folder, which is the folder where 11ty saves output files.
+* `.eslintrc.json` - EsLint Configuration data.
+* `.eslintignore` - EsLint exempted files.
+* `.nojekyll` - Required to let Github know that the site does not use Jekyll.
```
fixed

colinbdclark commented on AUTHORS.md:
```diff
@@ -0,0 +1,17 @@
+# Authors
+
+## Copyright Holders
+
+This is the list of FLOE Project copyright holders. It does not list all individual contributors because some have assigned
+copyright to an institution, only made minor changes, or their contributions no longer appear in the codebase.
+Please see the version control system's revision history for details on contributions.
+
+* OCAD University
+
+## Contributors
+
+Individual contributions can be viewed on the
+[Contributors](https://github.com/fluid-project/floeproject.org/graphs/contributors) page, or through the version control
+system's revision history.
+
+_**Note**: Individual authors may not hold copyright. See above "Copyright Holders" section for more information._
```
Our [standard license combination](https://wiki.fluidproject.org/display/fluid/Fluid+Licensing) these days is BSD-3 for source code and CC-BY International 4.0 for content. There's license boilerplate etc. on that wiki page as well. Of course, we have to declare copyright in order to appropriately license it.

jhung commented:
@sachin10101998 

1. The news index should be `./news/index.html` instead of `./newsindex.html`.

2. I was thinking the `permalink` front matter may not be necessary. Instead the navigation structure can be generated from the directory structure and the filename. This way it is easier to maintain and less prone to breaking.

3.  I tested removing HTML markup and replacing it with markdown in a few news articles and it worked fine. Some articles still required custom HTML markup (for example, `2019-11-05-Design-2-Align.md` you would keep the markup for `<figure>` and `<ul class="floe-news-inlineList">`).

sachin10101998 commented:
@jhung 
1)Fixed the news/index link.
2) permalinks are used to set the URL at which pages would be available as per convenience and are used to remove the day time stamp from the news index page. I tried to remove the permalink from one of the pages and then folders were being created for each of the post inside the news folder and each of those folders had an index.html page. I think the current way of defining the URLs makes more sense as we can classify if some particular page is a news article or not by looking at the URL and thereby it's easier to now the location of the file. 
3) I tried to change the p tags to simple paragraphs of markdown and it rendered in a shorter size compared to the original one. This is expected because even the p tags have some style statements related to them defined in foundation.css. Currently, I have changed the extensions of all md files to HTML and thus it makes more sense to keep using the HTML tags.

jhung commented:
@sachin10101998 

1. Great!

2. I'm looking into this into more detail. I'll get back to you on this.

3. Below is a screen capture from Firefox on Windows 10. Left is rendered from Markdown. Right is rendered from HTML source files. The markdown output looks acceptable to me. Is this not what you get in your instance?

![image](https://user-images.githubusercontent.com/564762/88689232-3d609d80-d0c8-11ea-9b94-4be7c55dc73a.png)

jhung commented:
@sachin10101998 

After some research, there seems to be a better way to create the News index and news permalinks. For example, if you checkout [this repo for the IDRC website,](https://github.com/inclusive-design/idrc) you can see a working example (live site is here: https://idrc.ocadu.ca/).

* ` idrc/src/posts/posts.json ` - post configuration for permalinks are generated from the path and article title.
* ` idrc/src/news.md` - defines configuration for news post index
* `idrc/src/_includes/layouts/news.njk` - index template for news posts

Using the IDRC site as an example, we can do similar here in Floe:

1. Define a `src/news/news.json` file that defines a permalink based on the directory path and the news filename.
2. Delete the News index in `src/news/`, and instead define a News index template and generate it from the collection of content in the `src/news/` directory.

Relevant 11ty documentation:
* https://www.11ty.dev/docs/data-template-dir/ - directory data for templates.

Please find me on #fluid-work IRC if you have any questions or comments.

sachin10101998 commented:
@jhung I have made the required changes.

jhung commented:
@sachin10101998 - please examine and test the output carefully. There are 3 copies of news posts being generated in the `.dist/` directory, and there are 2 indexes for news content. This is not what is expected.

The news content should all be in the `./news/` directory (not in sub-directories), with a single `./news/index.html` to list them.

sachin10101998 commented:
>  please examine and test the output carefully. There are 3 copies of news posts being generated in the .dist/ directory, and there are 2 indexes for news content. This is not what is expected.

@jhung , this is really strange. I did a fresh build and I don't see any duplicate copies. 
![Screenshot from 2020-08-07 22-49-16](https://user-images.githubusercontent.com/34926285/89671369-7d155980-d900-11ea-9ba0-54ecb7d7df99.png)

> The news content should all be in the ./news/ directory (not in sub-directories), with a single ./news/index.html to list them.

Done.

jhung commented:
Looking better @sachin10101998!

I was looking at the News filenames and thinking about a strategy that works better long-term. Here's what I'm thinking:

The default output filename for News posts should be the `{article date}-{article title}.html`.

But, News posts that have already been published don't strictly follow this scheme. In this case we can set a flag in the front matter `LegacyFilename: true` so the output filename would be `{article filename}.html`.

So to summarize, any new News article created in the future would use the `{article date}-{article title}.html`, and any old legacy News article would use `{article filename}.html` if an appropriate flag is set to `true` in the front matter.

Does this strategy sound okay? @BlueSlug , any thoughts?

sachin10101998 commented:
@jhung wouldn't it be better to have a common filename structure for all posts for consistency? We can change the output file name for old posts too since we have the dates on which they were published.

BlueSlug commented on src/resources.html:
```diff
@@ -0,0 +1,334 @@
+---
+layout: layouts/default
+title: Resources | floe
```
Please remove `| floe` from the title in this file, that should be added in the template since every page has this in its title

sachin10101998 replied:
fixed

sachin10101998 commented on src/_data/README.md:
```diff
@@ -0,0 +1,2 @@
+**_data***
```
fixed

sachin10101998 commented on src/_data/navigation.json:
```diff
@@ -0,0 +1,29 @@
+{
+    "items": [
+        {
+        "text": "Home",
```
fixed

sachin10101998 commented on src/_includes/partials/footer.html:
```diff
@@ -0,0 +1,32 @@
+            <div>
+                <p>
+                    Floe is a project of the
+                    <a href="http://idrc.ocadu.ca">Inclusive Design Research Centre</a>, <a href="http://ocadu.ca">OCAD University</a>.<br/>
```
fixed

sachin10101998 commented on src/_includes/partials/uio.html:
```diff
@@ -0,0 +1,28 @@
+<!-- BEGIN markup for Preference Editor -->
```
Removing it makes sense. Removing it for now. Will ad if @jhung thinks otherwise.

jhung replied:
I agree. This comment is no longer needed since uio is in a partial now.

sachin10101998 commented on src/accessibilitySprint2015.html:
```diff
@@ -0,0 +1,331 @@
+---
+layout: layouts/default
+title: OER ACCESSIBILITY SPRINT | floe
```
fixed

sachin10101998 commented on Gruntfile.js:
```diff
@@ -0,0 +1,136 @@
+        lintAll: {
+            sources: {
+                json: ["package.json", ".eslintrc.json","./src/_data/*.json"],
+                js: ["./src/transforms/*.js","./src/filters/*.js","./src/assets/js/floe.js","./src/assets/js/url.js","./src/assets/js/resources.js","./src/filters/*.js","./src/utils/*.js",".eleventy.js","Gruntfile.js"],
```
fixed

sachin10101998 commented on Gruntfile.js:
```diff
@@ -0,0 +1,136 @@
+            sources: {
+                json: ["package.json", ".eslintrc.json","./src/_data/*.json"],
+                js: ["./src/transforms/*.js","./src/filters/*.js","./src/assets/js/floe.js","./src/assets/js/url.js","./src/assets/js/resources.js","./src/filters/*.js","./src/utils/*.js",".eleventy.js","Gruntfile.js"],
+                md: ["./src/news/*.md"]
```
Fixed. I didn't add lib files because it's being copied from node modules and we decided to lint only third party code and not our own infusion code.

sachin10101998 commented on src/assets/stylesheets/foundation/README.md:
```diff
@@ -7,3 +7,4 @@
   * Grid max-width was set to 76rem
```
fixed

sachin10101998 commented on src/filters/date-filter.js:
```diff
@@ -0,0 +1,17 @@
+/* eslint-env es6 */
+/* eslint-disable no-console */
```
fixed

sachin10101998 commented on src/filters/markdown-filter.js:
```diff
@@ -0,0 +1,12 @@
+/* eslint-env es6 */
+/* eslint-disable no-console */
```
fixed

sachin10101998 commented on src/filters/w3-date-filter.js:
```diff
@@ -0,0 +1,8 @@
+/* eslint-env es6 */
+/* eslint-disable no-console */
```
fixed

sachin10101998 commented on src/filters/date-filter.js:
```diff
@@ -0,0 +1,17 @@
+    const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+    const dayWithSuffix = appendSuffix(dateObject.getDate());
+
+    return `${dayWithSuffix} ${months[dateObject.getMonth()]} ${dateObject.getFullYear()}`;
```
fixed

sachin10101998 commented on src/newsindex.md:
```diff
@@ -0,0 +1,16 @@
+<div class="floe-news-archive">
+<ul >
+{%- for post in collections.post reversed -%}
+<li><a class="hreff" href="{{ post.url }}"><p> {{ post.data.title }}</p></a>
```
It was earlier used for setting permalinks. I found a workaround for that so it's not used anymore. Removed it.

sachin10101998 commented on src/newsindex.md:
```diff
@@ -0,0 +1,16 @@
+---
+layout: layouts/post
```
fixed

sachin10101998 commented on src/news/2013-03-05-fluid-releases-new-video-player.md:
```diff
@@ -0,0 +1,32 @@
+The Fluid community has announced the release of Video Player v0.1
+
+Video Player is designed to be a fully accessible HTML5 video player. The designs can be seen at
+</br/>
```
fixed

BlueSlug replied:
This is still not fixed. Again, please make sure to check the output before pushing it to GitHub.

sachin10101998 commented on src/news/2014-03-10-harnessing-the-open-web.md:
```diff
@@ -0,0 +1,12 @@
+---
+Last week, the Floe team presented a paper entitled
+"Harnessing the Open Web To Create Inclusive Learning Environmen
+ts" at
```
fixed

sachin10101998 commented on src/news/2015-05-08-workshop-business-advantages.md:
```diff
@@ -0,0 +1,29 @@
+The [National Business and Disability Council (NBDC)](http://www.viscardicenter.org/services/nbdc/)
+at [The Viscardi Center](http://www.viscardicenter.org/)
+is offering an educational online workshop to focus on designing for diversity.
+No one needs to point out that the current business climate is in a state of flux
```
fixed

sachin10101998 commented on src/prefsEditors.html:
```diff
@@ -0,0 +1,26 @@
+---
+layout: layouts/default
+title: Preference Editing Tools | floe
```
fixed

sachin10101998 commented on src/newsindex.md:
```diff
@@ -0,0 +1,16 @@
+<ul >
+{%- for post in collections.post reversed -%}
+<li><a class="hreff" href="{{ post.url }}"><p> {{ post.data.title }}</p></a>
+<time class="floe-date" datetime="{{ post.data.date | w3DateFilter }}">{{ post.data.date | dateFilter }}</time>
```
This is really strange. The dates are in sync for me. How do I resolve this?

sachin10101998 commented:
@BlueSlug I have addressed all review points except the one where the dates in the news index page are rendering on the basis of the time zone in which a person is in. This is really strange because the dates are in sync for me and the dates are anyway being rendered from the hardcoded front matter. 
![Screenshot from 2020-08-09 22-06-28](https://user-images.githubusercontent.com/34926285/89737179-a4992d00-da8c-11ea-8e11-9adca33e918b.png)

jhung commented:
@sachin10101998, @BlueSlug 
Thanks for the comments re: file names for news posts.

The requirement is that any old URLs for News posts should still be valid on the Eleventy version of the site.

The problem here is that the old filenames do not all follow the same filename convention, which is why this can not be automatically generated from date and title of the article. For example, one article has the title: `Updates to the Inclusive Learning Design Handbook`, but it's original filename is `2012-02-28-updates-to-ildh.html`. 

This means that for the old articles that don't follow the new desired format, we still need to maintain its old filename somewhere so we know how to generate its output.

For new articles, we can automatically generate the filename, but for old articles we will need a special process added so that the original filename is output on generation. My suggestion was to add the `LegacyFilename` to the frontmatter, but I'm open to other ideas.

(I'm still catching up on the comments from above).

jhung commented:
@sachin10101998 
There are news posts that do not match the original filename:

* Files with incorrect dates. Example: `2012-03-26-webinar-outside-in.html` should be `2012-03-06-webinar-outside-in.html`.

* Files with incorrect capitalization. Example: `2016-11-30-iidex-2016.html` should be `2016-11-30-IIDEX-2016.html`.

* Files with character encoding artifacts? Example: `2014-03-28-workshopand58-inclusive-technology-in-the-classroom.html` should be `2014-03-28-workshop-inclusive-technology-in-the-classroom.html`

There are at least 12 files that don't match the original, and there may be some more.

jhung commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+
+**Contributions to the project can be made in following ways:**
+
+1. *Contribution towards building features for the project.*
+
+* There are several features that still require attention.
+  * Improving the User interface of submitted answers.
+  * Search functionality among the site data.
+  * Support for Comments.
+
```
This is not required and can be deleted.

sachin10101998 replied:
fixed

jhung commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+1. *Submitting new blog posts or news articles*
+
+* Post submission through Github:
+  * Go to the `src/news` folder
+  * Create a new Markdown file with title in the following format - `YYYY-MM-DD-titleOfNewsArticle.md`.
+  * Use the YYYY-MM-DD-newsarticlename.md file as template.
+  * Update the date, title and permalink in the front matter.
+  * Your article must have a specific template as mentioned below to get merged. make sure to follow the
+    template while submitting the answer.
+  * The template is as follows:
+   `---`
+   `layout: layouts/post`
+   `title: Title of News Article`
+   `date: 'YYYY-MM-DD'`
+   `tags: post`
+   `permalink: news/YYYY-MM-DD-newsArticleName.html`
+   `---`
+   `Content Data`
```
This documentation get updated including documenting how output filenames are generated from the `date`, and `title` or `filename` value

Also should document that `filename` should be used in situations where the title may be too long and a shorter filename is desirable. It is also used to ensure consistency for legacy news posts.

sachin10101998 replied:
fixed

jhung commented on package.json:
```diff
@@ -0,0 +1,50 @@
+    "name": "floeproject.org",
+    "version": "1.0.0",
+    "description": "FLOE Project Website",
+    "scripts": {
```
Add a `clean` script that removes node modules and dist.

For an example, please see: https://github.com/inclusive-design/idrc/blob/01d005b0483e5f7654c291bfbf508a5a2de46248/package.json#L12

sachin10101998 replied:
fixed

jhung commented on package.json:
```diff
@@ -0,0 +1,50 @@
+    "description": "FLOE Project Website",
+    "scripts": {
+        "build": "eleventy",
+        "eleventy": "eleventy --serve",
+        "eleventyport": "eleventy --serve --port=9778",
```
Why are there two scripts to start eleventy? Are both needed?

Also, to be consistent with other IDRC web projects, I would like to propose that `eleventyport` be replaced with `start` (see below).

For example, in our web projects we have:
* `clean` - deletes node_modules and output / build directories.
* `build` - deletes any existing (old) output files, and generates the output.
* `install` - installs dependencies
* `start` - runs clean, build, then runs the server that watches for changes in the output

Examples:
* https://github.com/inclusive-design/idrc/blob/dev/package.json
* https://github.com/platform-coop-toolkit/pinecone/blob/dev/package.json

sachin10101998 replied:
The default port at which eleventy runs is 8080. the eleventyport script is for running it at a different port i.e 9778.
I've made the required changes.

sachin10101998 commented on src/_includes/partials/footer.html:
```diff
@@ -0,0 +1,32 @@
+                <p>
+                    <span class="floe-hewlett-logo">Funded by a grant from <a href="https://www.hewlett.org">The William and Flora Hewlett Foundation</a>.</span>
+                </p>
+            </div>
```
fixed

jhung commented:
@sachin10101998 there are still filenames that do not match the original naming.

Please reference the original filenames here (https://github.com/fluid-project/floeproject.org/tree/master/news) and compare with the output in the `dist/news/` directory.

sachin10101998 commented:
@jhung made the changes.

jhung commented:
@sachin10101998 
The following 3 files need to be fixed.

`2012-05-12-floe-leads-inclusive-design-workshop-at-big-ideas-fest.html` should be:
`2012-12-05-floe-leads-inclusive-design-workshop-at-big-ideas-fest.html`

`2016-11-30-DEEP-Conference-2016.html` should be:
`2016-11-30-DEEP-conference-2016.html`

`2016-11-30-UN-International-Day-of-Persons-with-Disabilities.html` should be:
`2016-11-30-UN-international-day-of-persons-with-disabilities.html`

sachin10101998 commented:
@jhung changed.

jhung commented on src/news/2019-08-12-Coding-Prototype-Co-design.md:
```diff
@@ -0,0 +1,20 @@
+[Coding for All: Instruction That Includes Students With Special Needs](https://conference.iste.org/2019/program/search/detail_session.php?id=112083158).
+
+<img src="images/CodingCamp.png" alt="Camper coding sequences to create art with paint,
+ markers and the Dash and Sphero robots." /></br/>
```
You can delete this `<br/>` tag.

sachin10101998 replied:
fixed

sachin10101998 commented on README.md:
```diff
@@ -1,20 +1,21 @@
+## To Build Locally
+
+1. Get the required node modules: `npm install`
+2. Run eleventy from the fluid-website directory `npm run eleventyport`.
+3. Open `http://localhost:9778/` to see the website.
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+  * Your article must have a specific template as mentioned below to get merged. make sure to follow the
+    template while submitting the answer.
+  * The template is as follows:
+   `---`
+   `layout: layouts/post`
+   `title: Title of News Article`
+   `date: 'YYYY-MM-DD'`
+   `tags: post`
+   `filename: Shorter URL. Can also contain capital characters`
+   `---`
+   `Content Data`
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+   `title: Title of News Article`
+   `date: 'YYYY-MM-DD'`
+   `tags: post`
+   `filename: Shorter URL. Can also contain capital characters`
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+    template while submitting the answer.
+  * The template is as follows:
+   `---`
+   `layout: layouts/post`
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+   `layout: layouts/post`
+   `title: Title of News Article`
+   `date: 'YYYY-MM-DD'`
+   `tags: post`
```
layout and tags can be removed. Removed them

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+   `---`
+   `Content Data`
+
+  * The URL for news articles can be generated from title and date or filename or date metadata in front matter.
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+   `Content Data`
+
+  * The URL for news articles can be generated from title and date or filename or date metadata in front matter.
+  * If filename metadata is present, then filename and date will be used for creating URL else title and date would be used.
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+
+  * The URL for news articles can be generated from title and date or filename or date metadata in front matter.
+  * If filename metadata is present, then filename and date will be used for creating URL else title and date would be used.
+  * The filename metadata is used for creating custom URLs.
+  * There is a bit of inconsistence in older blog filenames. Some old news articles have capital
+    characters in their URL and hence 'filename' metadata has been set in front matter of such files.
+  * You need to set 'filename' metadata in front matter only if the title of your news article is quite long and
+    you need a shorter URL for the post thereby creating a custom URL as mentioned above.
```
fixed

sachin10101998 commented on src/index.html:
```diff
@@ -0,0 +1,187 @@
+                    for the full diversity of learners, leveraging the variants made possible by
+                    <a href="http://www.hewlett.org/programs/education/open-educational-resources">Open Education Resources</a> (OER).</p>
+                <p>FLOE is led by the <a href="http://idrc.ocadu.ca">Inclusive Design Research Centre</a>
+                    and applies <a href="http://idrc.ocad.ca/index.php/resources/idrc-online/library-of-papers/443-whatisinclusivedesign">Inclusive Design</a> to open learning.</p>
```
fixed

sachin10101998 commented on src/uioptions.html:
```diff
@@ -0,0 +1,34 @@
+---
+layout: layouts/default
+title: UI Options
+permalink: uioptions.html
```
fixed

sachin10101998 commented on src/_data/navigation.json:
```diff
@@ -0,0 +1,29 @@
+        },
+        {
+            "text":"UI Option",
+            "url":"/uioptions.html",
```
fixed

sachin10101998 commented on src/_includes/layouts/default.html:
```diff
@@ -0,0 +1,37 @@
+        {% if page.url == "/resources.html" %}
+        <script type="text/javascript" src="{{ '/' | url }}assets/js/resources.js"></script>
+        {% endif %}
+        {% if page.url == "/uioptions.html" %}
```
fixed

sachin10101998 commented on src/_includes/partials/header.html:
```diff
@@ -0,0 +1,20 @@
+            <ul>
+                <li><a href="{{ '/' | url }}./news/index.html">News</a></li>
+                <li><a href="{{ '/' | url }}./resources.html">Resources</a></li>
+                <li><a href="{{ '/' | url }}./uioptions.html">UI Options</a></li>
```
fixed

sachin10101998 commented on src/index.html:
```diff
@@ -0,0 +1,187 @@
+                and tools to the design process.
+                </p>
+
+                <p>The <a href="http://metadata.floeproject.org/demos/metadata/">Metadata Editor</a>
+                makes it simple to provide useful labels regarding the learner preferences the resource can meet,
+                and supports authors in adding accessibility features.
+                The <a href="http://metadata.floeproject.org/demos/feedback/">Feedback Tool</a>
+                allows users to help in the refinement of content and its metadata.</p>
```
fixed

sachin10101998 commented on src/index.html:
```diff
@@ -0,0 +1,187 @@
+                    <dt><a href="http://www.oercommons.org">OER Commons</a></dt>
+                    <dd>OER Commons has integrated Learner Options across their entire site and has also integrated the Floe designs for content authoring in Open Author.</dd>
+                    <dt><a href="http://oerpub.org/">OERPUB</a></dt>
+                    <dd>OERPUB is using the Floe Video Player and Metadata Author to create rich, customizable learning content.</dd>
```
fixed

jhung commented on src/index.html:
```diff
@@ -0,0 +1,184 @@
+                <p>Floe has added a <a href="https://github.com/fluid-project/infusion-docs/blob/master/src/documents/TextToSpeechAPI.md">text-to-speech component</a>
+                to the Infusion JavaScript framework. The component uses browser-based speech synthesis
+                to provide self-voicing capabilities to web resources.</p>
+                <p>The<a href="https://github.com/fluid-project/metadata)">source code for the Metadata Editor and Feedback Tool</a>is available on Github.</p>
```
Instead of a new paragraph, you can just add the text to the end of line 123.

sachin10101998 replied:
fixed

jhung commented:
@BlueSlug I think this PR is ready for a look-over.

BlueSlug commented on src/news/2019-10-23-SJRK-F2F.md:
```diff
@@ -0,0 +1,17 @@
+with youth at the margins, enabling them to not only tell their stories but
+also to capture and present them in ways they want to.
+
+<img src="images/SJRK_Drawing.png" alt="Collaborative surprise drawings by the SJRK team." /></br/>
```
Please fix the `br` element, it's rendering as "</br/>" in the page content

BlueSlug commented on src/news/2019-07-02-Guelph-accessibility-conference.md:
```diff
@@ -0,0 +1,21 @@
+
+<img src="images/guelph-2019-jt.png" alt="Photograph from a distance of Jutta Treviranus delivering a keynote." /></br/>
+
+### Jutta Treviranus's opening keynote slides
```
This is supposed to be italicized text, not a heading

BlueSlug commented on src/news/2019-07-02-Guelph-accessibility-conference.md:
```diff
@@ -0,0 +1,21 @@
+titled "From Intelligence to Wisdom: Navigating a Complex Future". Her talk covered a range of topics from
+changing educational practice to the future of artificial intelligence.
+
+<img src="images/guelph-2019-jt.png" alt="Photograph from a distance of Jutta Treviranus delivering a keynote." /></br/>
```
"</br/>" is appearing on the page as text, please fix this element

BlueSlug commented on src/_includes/layouts/post.html:
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+    <head>
+        <title>{% if title %} {{ title }} {% else %} {{ site.title }} {% endif %} | floe</title>
```
The page title should begin with "News: "

BlueSlug commented on src/news/2019-03-22-OCAD-OE-week.md:
```diff
@@ -0,0 +1,35 @@
+date: '2019-03-22'
+---
+Members of the Floe team recently participated in [OCAD University's](https://www.ocadu.ca/)
+[Open EducationWeek](https://www.openeducationweek.org/) event,
```
Please add a space between "Education" and "Week"

BlueSlug commented on src/news/2019-03-22-OCAD-OE-week.md:
```diff
@@ -0,0 +1,35 @@
+three blocks of text: 'Understanding: make it accessible and clear', 'Sharing: make it easy
+to adapt and share', 'Participation & Inclusion: make it inviting, safe and sustainable.'" /></br/>
+
+**One of panelist Jairus Khan's slides, about elements of openness.**
```
This is meant to be italicized, not bold text. Please also remove the line above it, since it's not meant to be a new paragraph

BlueSlug commented on src/news/2019-03-20-NROC-2019.md:
```diff
@@ -0,0 +1,18 @@
+over the United States. This year's theme was called "A Thriving Hive" - focusing
+on collaboration and mutual success through our unique contributions.
+
+The IDRC presented "Accessibility Through Inclusive Design Thinking
```
Please add `"` at the end to close the quotations

BlueSlug commented on src/news/2019-03-20-NROC-2019.md:
```diff
@@ -0,0 +1,18 @@
+on collaboration and mutual success through our unique contributions.
+
+The IDRC presented "Accessibility Through Inclusive Design Thinking
+[PDF of slides](https://wiki.fluidproject.org/download/attachments/1707985/NROC%20Member%20Meeting%202019%20-%20Accessibility%20through%20inclusion.pdf?version=1&modificationDate=1552341970417&api=v2)
```
Please add parentheses around the link to match the original content

BlueSlug commented on src/news/2018-09-17-GSoC2018.md:
```diff
@@ -0,0 +1,97 @@
+
+#### [Import external data to MyL3](https://summerofcode.withgoogle.com/archive/2018/projects/5924252133883904/)
+
+#### by Jeevan Prakash
```
Please put this text at the end of the heading above:
```
#### [Import external data to MyL3](https://summerofcode.withgoogle.com/archive/2018/projects/5924252133883904/) by Jeevan Prakash
```

And please do the same for all project titles in this article

BlueSlug replied:
The first three project titles look great! This is still an issue for the last two ("Playtown!" and "Upgrading Achecker")

sachin10101998 replied:
fixed

BlueSlug commented on src/news/2018-06-19-El-Planeta-Es-La-Escuela.md:
```diff
@@ -0,0 +1,50 @@
+
+<figure>
+<a href="images/Medellin_art.png"><img src="images/Medellin_art_thumb.png" alt="view larger version of this image"
+aria-details="det1">]
```
Please fix the link, it's mixing HTML and Markdown syntax

BlueSlug commented on src/news/2018-04-18-New-Content-Added-to-Inclusive-Design-Resources.md:
```diff
@@ -0,0 +1,45 @@
+
+<figure>
+<a href ="images/collaborative_art.png"><img src="images/collaborative_art_thumb.png"
+alt="view larger version of this image" aria-details="det1">]()
```
Please fix the link, it's mixing HTML and Markdown syntax. You should be careful of using search and replace without checking the results afterwards.

BlueSlug commented on src/news/2018-01-31-uioPlus.md:
```diff
@@ -0,0 +1,44 @@
+[OCAD University](https://www.ocadu.ca/) and the [Trace R&D Center](https://trace.umd.edu/) at [University of Maryland](https://umd.edu/)
+under funding for the [FLOE Project](https://floeproject.org/) from the [William and Flora Hewlett Foundation](https://www.hewlett.org/)
+and the [National Institute on Disability, Independent Living, and Rehabilitation Research (NIDILRR)](https://www.acl.gov/about-acl/about-national-institute-disability-independent-living-and-rehabilitation-research)
+,[Administration for Community Living](https://www.acl.gov/) under grant #90RE5027.
```
the comma at the beginning of this line should have a space after it, and not before it

BlueSlug replied:
The space before the comma is still present, since it's on a new line

sachin10101998 replied:
fixed

BlueSlug commented on src/news/2017-08-30-self-assessment-toolkit.md:
```diff
@@ -0,0 +1,78 @@
+(select to enlarge):
+
+<figure>
+<a href="images/Self-reflection-Figure-1.png"><img src="images/Self-reflection-Figure-1
```
None of the images in this article are showing, please fix them.

BlueSlug replied:
They are definitely not working for me. When I try to load the image on its own, I get the following error in my browser's console:
`GET http://localhost:8080/news/images/Self-reflection-Figure-1-thumb.png net::ERR_INVALID_CHUNKED_ENCODING 200 (OK)`

I believe this is happening because the URLs for the images in the markdown have dashes between each word (`Self-reflection-Figure-1-thumb.png`), but the files in the `dist\news\images` directory have spaces in their names (`Self-reflection Figure 1 thumb.png`)

sachin10101998 replied:
fixed

BlueSlug commented on src/news/2016-11-30-DEEP-conference-2016.md:
```diff
@@ -0,0 +1,21 @@
+date: '2016-11-30'
+---
+
+On October 13-14, 2016 the [DEEP -Discovery conference](http://deep.idrc.ocadu.ca)
```
Please add a space after the dash in the link text "Deep - Discovery conference"

BlueSlug commented on src/news/2016-11-30-IIDEX-2016.md:
```diff
@@ -0,0 +1,13 @@
+---
+
+The IDRC participated in the IIDEX National Summit on Accessibility on November 30th
+, 2016 ([2016 IIDEX Seminar Schedule](https://tbs.iidexcanada.com/conference/seminars2016)).
```
This comma should be on the line above, an extra space is being added in error

BlueSlug commented on src/news/2016-03-28-upcoming-talks-on-inclusive-digital-education.md:
```diff
@@ -0,0 +1,34 @@
+IDRC Director Jutta Treviranus will be speaking at the following conferences
+on topics crossing digital inclusion in education, to rethinking the role of technology
+in our institutions.
+April 2016:
```
Please add a blank line above this one...

BlueSlug commented on src/news/2016-03-28-upcoming-talks-on-inclusive-digital-education.md:
```diff
@@ -0,0 +1,34 @@
+in our institutions.
+April 2016:
+
+- April 12 -[WWW2016 Workshop onWeb Science and Technology for Education (WWW2016 Montreal)](https://events.keep.edu.hk/webed/2016/)
```
The formatting of this list is incorrect, please make it consistent with the original page

BlueSlug commented on src/news/2016-03-28-upcoming-talks-on-inclusive-digital-education.md:
```diff
@@ -0,0 +1,34 @@
+- May 12 to 14 - [UNIversal Inclusion: Rights and Opportunities for Persons with Disabilities in the Academic Context](https://www.unito.it/gallerie/universal-inclusion-rights-and-opportunities-persons-disabilities-academic-context)
+
+- The information about this conference is also available in this accessible PDF
+: [http://en.unito.it/sites/sten/files/universal_inclusion.pdf](http://en.unito.it/sites/sten/files/universal_inclusion.pdf)
```
The colon should be at the end of the line above

BlueSlug commented on src/news/2015-11-16-phet.md:
```diff
@@ -0,0 +1,27 @@
+---
+Floe is working with [PhET](http://phet.colorado.edu/)
+to improve the accessibility of their free, open, interactive physics simula
+tions. Over the past
```
Please fix the line breaks in this article so no words are being split in half, it's adding many unnecessary spaces

BlueSlug commented on src/news/2015-03-26-uio-wp-plugin.md:
```diff
@@ -0,0 +1,13 @@
+[Learner Options](http://build.fluidproject.org/infusion/demos/uiOptions/).
+The plugin adds the Learner Options component to any WordPress site, allowing visitors to adjust
+things like text size, text style, contrast, and line spacing.
+The Learner Options plugin is still in early development, but it has been deployed on
```
Please add a new line above this one...

BlueSlug commented on src/news/2015-03-26-uio-wp-plugin.md:
```diff
@@ -0,0 +1,13 @@
+things like text size, text style, contrast, and line spacing.
+The Learner Options plugin is still in early development, but it has been deployed on
+[this blog.](http://tjbliss.org/a-one-size-fits-one-experience-on-tjbliss-org/)
+The plugin is [available on GitHub](https://github.com/fluid-project/uio-wordpress-plugin/),
```
Please add a new line above this one...

BlueSlug commented on src/news/2015-01-13-oer-a11y-sprint.md:
```diff
@@ -0,0 +1,21 @@
+sign and develop tools and exemplars that make practical progress in the
+important agenda of inclusive learning through OER.
+They will be [sharing their results in open notebooks](https://docs.google.com/document/d/1EPdECgqM7BxDmfh5CHJ7ZYr00ldxUqkcnJOddaMvXTE/edit)
+Among the challenges on the agenda are a special focus on HTML5 and EPU
```
Please add a new line above this one...

BlueSlug commented on src/news/2014-09-30-lumen-learning.md:
```diff
@@ -0,0 +1,18 @@
+---
+[Lumen Learning](http://lumenlearning.com/) has been selected to receive funding as part of the
+[Next Generation Courseware Challenge](https://www.gatesfoundation.org/Media-Center/Press-Releases/2014/09/Gates-Foundation-Announces-Finalists-for-$20-Million-in-Digital-Courseware-Investments),
+a $20 million grant competition from the [Bill & Melind a Gates Foundation](http://www.gatesfoundation.org/)
```
Please fix Melinda's name

BlueSlug commented on src/news/2014-09-30-lumen-learning.md:
```diff
@@ -0,0 +1,18 @@
+educational resources. The result will be “next generation” courseware for four introductory-level
+college courses that enroll millions of students every year: Introduction to Business; Principles of
+Marketing; Microeconomics; and Macroeconomics.
+The Floe team is excited to be a partner in this project. Floe will design the courseware’s universally
```
Please add a new line above this one...

BlueSlug commented on src/news/2014-03-28-workshop-inclusive-technology-in-the-classroom.md:
```diff
@@ -0,0 +1,11 @@
+Earlier this week, the Floe team conducted a workshop with the
+[Vitrine technologie-éducation (VTÉ)](http://www.vteducation.org/en/articles/step/information-and-communication-technology-integration-post-secondary-education).
+The workshop discussed the use of technology to create an inclusive cla
+ssroom, and highlighted the work of the Floe project.
```
Please fix the line breaks so no words are split up

BlueSlug commented on src/news/2014-03-24-floe-presentations-at-csun.md:
```diff
@@ -0,0 +1,12 @@
+---
+Last week, the Floe team presented two papers at CSUN's annual
+[Internation Technology &amp; Persons With Disabilities conference](http://www.csun.edu/cod/conference/2014/sessions/index.php/public/website_pages/view/1)
+: "Fluid Infusion: Building the Next Gen
```
Please fix the colon so there's no space before it. If the linter is complaining about the line length, you can add an exception for this case

BlueSlug commented on src/news/2014-03-18-metadata-editing-components.md:
```diff
@@ -0,0 +1,16 @@
+bility metadata.
+These tools will allow content creators to easily generate metadata at the point of creation,
+reducing the cost and complexity of pro-ducing and delivering content that can be tailored to a user’s needs and preferences.
+A [demo](https://metadata.floeproject.org/demos/metadata/index.html)
```
Please add a new line above this one...

BlueSlug commented on src/news/2014-03-18-metadata-editing-components.md:
```diff
@@ -0,0 +1,16 @@
+The Floe team is developing a suite of tools for cr
+eating more accessible and
+personalizable web content by supporting the creation of accessi
+bility metadata.
```
Please ensure no words are split up

BlueSlug commented on src/news/2013-03-05-fluid-releases-new-video-player.md:
```diff
@@ -0,0 +1,32 @@
+
+For a complete list of known issues, please see the Video Player issue tracker.
+</br>
+
```
Please remove this blank line

BlueSlug commented on src/news/2012-12-05-floe-deployment-feasibility-study-funded.md:
```diff
@@ -0,0 +1,22 @@
+[IBM](http://www.ibm.com/us/en/),
+and [Inclusive Technologies](http://inclusive.com/).
+
+For more information about the grant,see
```
Please add a space before "see"

BlueSlug commented on src/news/2012-12-05-floe-leads-inclusive-design-workshop-at-big-ideas-fest.md:
```diff
@@ -0,0 +1,12 @@
+---
+
+On Dec. 3, 2012, Jutta Treviranus led a workshop on Inclusive Learning Des
+ign at Big Ideas 2012. The workshop highlighted techniques for creating inclusive open educational resources.
```
Please make sure no words are split across lines

BlueSlug replied:
There are still words in this file that are split up:

> Big Ideas is an annual conference that brings together teachers, policy **mak ers**, researchers, eduprenuers, students, and funders who are championing **th e** transformation of K-20 education. For information, visit http://www.bigideasfest.org/fests

BlueSlug commented on src/news/2012-10-24-jutta-zoomer-mag.md:
```diff
@@ -0,0 +1,17 @@
+Jutta Treviranus, Director of the
+[Inclusive Design Research Centre](http://idrc.ocadu.ca), has been named by
+[Zoomer Magazine](http://www.zoomermag.com/) as one of
+[Canada's Top 45 Over 45](http://www.everythingzoomer.com/arts-entertainment/stars-royals/2012/10/05/the-3rd-annual-zoomer-list-canadas-top-45-over-45-jutta-treviranus/)
```
Please add a comma after this link. If the linter complains about it, you can add an exception in this case.

BlueSlug commented on src/news/2012-10-19-plans-for-floe-year-three.md:
```diff
@@ -0,0 +1,27 @@
+work we've been doing on personalized learning. During the first two years of the Floe project, we
+
+- developed the [Inclusive Learning Design Handbook (ILDH)](http://handbook.floeproject.org/);
+
```
Please remove the empty lines between all the list items, it's putting the text into paragraphs which is incorrect

BlueSlug commented on src/news/2019-11-05-Design-2-Align.md:
The issue with turning it into a heading is that assistive technologies like screen readers will interpret it as a heading, which could be very confusing to users. The HTML solution works! 👍

BlueSlug commented on src/assets/js/floe.js:
```diff
@@ -1,29 +1,20 @@
+https://github.com/fluid-project/fluidproject.org/master/AUTHORS.md.
 */
 
 // Declare dependencies
```
It's still present

sachin10101998 replied:
Does the declare dependencies line needs to be removed?

sachin10101998 commented on src/filters/date-filter.js:
```diff
@@ -1,13 +1,11 @@
+    var temp1 = `${months[dateObject.getMonth()]}`;
+    var temp2 = ` ${dayWithSuffix}`;
+    var temp3 = ` ${dateObject.getFullYear()}`;
+    return temp1 + temp2 + "," + temp3;
```
fixed

sachin10101998 commented on src/news/2014-09-30-lumen-learning.md:
```diff
@@ -4,14 +4,15 @@ date: '2014-09-30'
 educational resources. The result will be “next generation” courseware for four introductory-level
 college courses that enroll millions of students every year: Introduction to Business; Principles of
-Marketing; Microeconomics; and Macroeconomics.
+Marketing; Microeconomics; and Macroeconomics..
```
fixed

BlueSlug commented:
Thanks for addressing those, @sachin10101998!

@jhung please consider this sentence my stamp of approval 👍

sachin10101998 commented on src/news/2015-11-16-phet.md:
```diff
@@ -4,18 +4,15 @@ filename: phet
-[Forces and Motion Basics](https://wiki.fluidproject.org/display/fluid/PhET+Forces+and+Motion+Simulation+Design)
+to improve the accessibility of their free, open, interactive physics simulations.
+Over the past year, we have contributed research, ideas, and evaluations for improvements
+to two of PhET's simulations:[Forces and Motion Basics](https://wiki.fluidproject.org/display/fluid/PhET+Forces+and+Motion+Simulation+Design)
```
fixed

jhung commented:
@gtirloni this is ready for CI testing. Thanks!
